synapse/storage/databases/main/event_push_actions.py

```diff
@@ -110,6 +110,8 @@ def _stream_ordering_from_event_id_and_room_id_txn(
 
         Returns:
             The stream ordering for this event.
+            We should always have a stream ordering to return, because the event ID
+            should come from a local read receipt.
```

Comment:
... unless that event has been purged?

Reply:
Ugh, true, I forgot about this possibility.